components/pages/explorer/RepoTable.tsx

```diff
@@ -229,19 +235,84 @@ const getTableStyle = (theme: typeof defaultTheme) => css`
 
+  const { NEXT_PUBLIC_ARRANGER_API_URL, NEXT_PUBLIC_FILE_DOWNLOAD_LIMIT } = getConfig();
   const today = new Date().toISOString().slice(0, 10).replace(/-/g, '');
+  const { selectedTableRows } = props;
+  const [seqFilesDownloading, setSeqFilesDownloading] = useState<boolean>(false);
+  const [seqFilesDownloadError, setSeqFilesDownloadError] = useState<string | null>(null);
+
+  const downloadFilesEnabled =
+    !seqFilesDownloading &&
+    selectedTableRows.length &&
+    selectedTableRows.length <= NEXT_PUBLIC_FILE_DOWNLOAD_LIMIT;
+
   const customExporters = [
-    { label: 'File Table', fileName: `data-explorer-table-export.${today}.tsv` }, // exports a TSV with what is displayed on the table (columns selected, etc.)
-    { label: 'File Manifest', fileName: `score-manifest.${today}.tsv`, columns: manifestColumns }, // exports a TSV with the manifest columns
+    { label: 'Export Table to TSV', fileName: `data-explorer-table-export.${today}.tsv` }, // exports a TSV with what is displayed on the table (columns selected, etc.)
+    {
+      label: () => (
+        <span
+          // TODO: disabled style will be fixed in theming ticket https://github.com/supercluster-covid-data-portal/ui/issues/1
+          css={(theme) => css`
+            cursor: ${downloadFilesEnabled ? 'pointer' : 'not-allowed'};
+            color: ${downloadFilesEnabled ? theme.colors.black : theme.colors.grey_5};
+          `}
+          onClick={
+            downloadFilesEnabled
+              ? async () => {
+                  setSeqFilesDownloading(true);
+                  ajax
+                    .post(
+                      urlJoin(NEXT_PUBLIC_ARRANGER_API_URL, DOWNLOAD_SEQ_PATH),
+                      {
+                        ids: selectedTableRows,
+                      },
+                      {
+                        responseType: 'blob',
+                        headers: { accept: '*/*' },
+                      },
+                    )
+                    .then((res: AxiosResponse) => {
+                      const blob = new Blob([res.data]);
+                      const filename = res.headers['content-disposition'].split('"')[1];
+                      createDownloadInWindow(filename, blob);
+                    })
+                    .catch(async (err: AxiosError) => {
+                      if (err) {
+                        const code = err.response?.status || 500;
+                        const text = await new Response(err.response?.data).text();
+                        const displayError = getDownloadError(code, text);
+                        return setSeqFilesDownloadError(displayError);
+                      }
+                      setSeqFilesDownloadError('An unknown error occurred. Please try again');
+                    })
+                    .finally(() => {
+                      setSeqFilesDownloading(false);
+                    });
+                }
+              : () => null
```

Comment:
There is so much code inline here, surely we want to move this element and its clickHandler into their own methods just for code organization purposes...